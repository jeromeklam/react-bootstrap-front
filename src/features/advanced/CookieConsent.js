import React from 'react';
import PropTypes from 'prop-types';

import { Container, Row, Col } from '../grid';

export default function CookieConsent(props) {
  console.log(props);
  return (
    <div className="advanced-cookie-consent">
      <Container size="xs" className="advanced-cookie-consent-inner">
        <Row>
          <Col size={36}>
            <h4>{props.title}</h4>
          </Col>
        </Row>
        <Row>
          <Col size={36}>
            <p style={{ fontSize: '0.8rem' }}>{props.text}</p>
          </Col>
        </Row>
        <Row>
          <Col size={36} className="text-right">
            {props.onDecline &&
              <button type="button" className="btn btn-warning" onClick={props.onDecline}>
                {props.decline}
              </button>
            }
            <button type="button" className="btn btn-primary" onClick={props.onAccept}>
              {props.accept}
            </button>
          </Col>
        </Row>
      </Container>
    </div>
  );
}

CookieConsent.propTypes = {
  accept: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  decline: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  onAccept: PropTypes.func.isRequired,
  onDecline: PropTypes.func,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
};
CookieConsent.defaultProps = {
  decline: "",
  onDecline: null,
};
