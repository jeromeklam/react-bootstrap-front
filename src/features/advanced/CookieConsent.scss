@import '../../styles/mixins';

.advanced-cookie-consent {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  opacity: 0.8;
  background-color: theme-color('secondary');
  color: white;
  padding: 10px;

  .advanced-cookie-consent-inner {
    max-width: 1024px;
    margin: 0 auto;
    padding-top: 10px;
  }
}

@media (min-width: 1024px) {
  .advanced-cookie-consent {
    height: 250px;
    top: auto;
  }
}