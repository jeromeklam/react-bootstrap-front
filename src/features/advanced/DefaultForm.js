import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const cardStyles = {
  position: 'absolute',
  top: '0px',
  bottom: '0px',
  left: '0px',
  right: '0px',
};

const cardBodyStyles = {
  overflowX: 'hidden',
  overflowY: 'auto',
};

export const DefaultForm = props => (
  <form className="layout-form-responsive">
    <div className="card bg-light" style={cardStyles}>
      <div className="card-header bg-secondary clearfix">
        <div className="float-right">
          <span className="navbar-brand text-light">{props.title}</span>
        </div>
        {props.tabs && props.tabs.length > 0 && (
          <ul className="nav nav-tabs float-left">
            {props.tabs &&
              props.tabs.map(oneTab => (
                <li key={oneTab.key} data-id={oneTab.key} className="nav-item">
                  <a
                    className={classnames('nav-link', props.tab === oneTab.key && 'active')}
                    onClick={() => {
                      props.onNavTab(oneTab.key);
                    }}
                  >
                    {oneTab.label}
                  </a>
                </li>
              ))}
          </ul>
        )}
      </div>
      <div className="card-body" style={cardBodyStyles}>{props.children}</div>
      <div className="card-footer text-right bg-secondary">
        <button type="button" className="btn btn-success btn-submit" onClick={props.onSubmit}>
          <span>Enregistrer</span>
        </button>
        &nbsp;
        <button type="button" className="btn btn-secondary" onClick={props.onCancel}>
          <span>Annuler</span>
        </button>
      </div>
    </div>
  </form>
);

DefaultForm.propTypes = {
  title: PropTypes.string.isRequired,
  tabs: PropTypes.element,
  tab: PropTypes.string,
  children: PropTypes.element.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  onNavTab: PropTypes.func,
};

DefaultForm.defaultProps = {
  tabs: [],
  tab: '',
  onNavTab: (() => {}),
};
