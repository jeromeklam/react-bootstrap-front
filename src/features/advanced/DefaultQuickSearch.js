import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class DefaultQuickSearch extends Component {
  static propTypes = {
    quickSearch: PropTypes.element.isRequired,
    onSubmit: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    label: PropTypes.string,
    icon: PropTypes.element.isRequired,
    plus: PropTypes.element,
    reset: PropTypes.element,
  };

  static defaultProps = {
    label: '',
    plus: null,
    reset: null,
  };

  static getDerivedStateFromProps(props, state) {
    if (props.quickSearch !== state.current) {
      return { current: props.quickSearch, search: props.quickSearch };
    }
    return null;
  }

  constructor(props) {
    super(props);
    this.state = {
      current: this.props.quickSearch,
      search: this.props.quickSearch,
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(event) {
    if (event) {
      event.preventDefault();
    }
    this.setState({ search: event.target.value });
  }

  onSubmit(event, mode) {
    if (event) {
      event.preventDefault();
    }
    this.props.onSubmit(this.state.search, mode);
  }

  render() {
    return (
      <form className="input-quick_search" onSubmit={this.onSubmit}>
        <div className="input-group">
          <input
            type="text"
            className="form-control"
            name={this.props.name}
            placeholder={this.props.label}
            value={this.state.search}
            onChange={this.onChange}
          />
          <div className="input-group-append">
            <button type="button" title="Recherche simple" className="btn btn-outline-secondary bg-light" onClick={(ev) => this.onSubmit(ev, "simple")}>
              {this.props.icon}
            </button>
            {this.props.plus &&
              <button type="button" title="Recherche mois stricte" className="btn btn-outline-secondary bg-light" onClick={(ev) => this.onSubmit(ev, "plus")}>
                {this.props.plus}
              </button>
            }
            {this.props.reset &&
              <button type="button" title="Initialiser la recherche" className="btn btn-outline-secondary bg-light text-warning" onClick={(ev) => this.onSubmit(ev, "reset")}>
                {this.props.reset}
              </button>
            }
          </div>
        </div>
      </form>
    );
  }
}
