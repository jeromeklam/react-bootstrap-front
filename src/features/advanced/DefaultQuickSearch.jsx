import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class DefaultQuickSearch extends Component {
  static propTypes = {
    quickSearch: PropTypes.element.isRequired,
    onSubmit: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    label: PropTypes.string,
    icon: PropTypes.element.isRequired,
  };

  static defaultProps = {
    label: '',
  };

  static getDerivedStateFromProps(props, state) {
    if (props.quickSearch !== state.current) {
      return { current: props.quickSearch, search: props.quickSearch };
    }
    return null;
  }

  constructor(props) {
    super(props);
    this.state = {
      current: this.props.quickSearch,
      search: this.props.quickSearch,
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(event) {
    if (event) {
      event.preventDefault();
    }
    this.setState({ search: event.target.value });
  }

  onSubmit(event) {
    if (event) {
      event.preventDefault();
    }
    this.props.onSubmit(this.state.search);
  }

  render() {
    return (
      <form className="input-quick_search" onSubmit={this.onSubmit}>
        <div className="input-group">
          <input
            type="text"
            className="form-control"
            name={this.props.name}
            placeholder={this.props.label}
            value={this.state.search}
            onChange={this.onChange}
          />
          <div className="input-group-append">
            <button type="button" className="btn btn-outline-secondary bg-light" onClick={this.onSubmit}>
              {this.props.icon}
            </button>
          </div>
        </div>
      </form>
    );
  }
}
