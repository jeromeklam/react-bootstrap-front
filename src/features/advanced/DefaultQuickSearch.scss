.input-quick_search {
  .form-control {
    border-color: theme-color('secondary');
  }
}