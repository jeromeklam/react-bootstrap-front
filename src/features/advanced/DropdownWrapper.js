import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { Dropdown } from '../basic';

export default class DropdownWrapper extends Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
    trigger: PropTypes.element.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = { open: false, ref: React.createRef() };
    this.onClose = this.onClose.bind(this);
    this.onClick = this.onClick.bind(this);
  }

  onClick(e) {
    if (e) {
      e.preventDefault();
      e.stopPropagation();
    }
    this.setState({ open: true });
  }

  onClose(e) {
    this.setState({ open: false });
  }

  render() {
    return (
      <>
        <div ref={this.state.ref} className="advanced-dropdown-wrapper" onClick={this.onClick}>
          {this.props.trigger}
        </div>
        {this.state.open && (
          <Dropdown myRef={this.state.ref} onClose={this.onClose}>
            {this.props.children}
          </Dropdown>
        )}
      </>
    );
  }
}
