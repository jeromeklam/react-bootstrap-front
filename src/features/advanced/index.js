export * from './DefaultForm.jsx';
export { default as DefaultQuickSearch } from './DefaultQuickSearch.jsx';
export { default as HoverObserver } from './HoverObserver.jsx';
export * from './MobileForm.jsx';
export { default as MobileQuickSearch } from './MobileQuickSearch.jsx';
export { default as Portal } from './Portal.jsx';
export * from './ResponsiveConfirm.jsx';
export * from './ResponsiveForm.jsx';
export * from './ResponsiveModal.jsx';
export { default as ResponsiveModalInner } from './ResponsiveModalInner.jsx';
export * from './ResponsiveQuickSearch.jsx';
export { default as WidthObserver } from './WidthObserver.jsx';
export { default as SvgMask } from './SvgMask';
export { default as Follower } from './Follower';
export { default as SvgTimer } from './SvgTimer';
export { default as SvgPlay } from './SvgPlay';
