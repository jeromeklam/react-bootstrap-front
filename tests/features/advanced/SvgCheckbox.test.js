import React from 'react';
import renderer from 'react-test-renderer';
import { SvgCheckbox } from '../../../src/features/advanced';

it('renders with className', () => {
  // First render
  const component = renderer.create(
    <SvgCheckbox className="btn bg-secondary" />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});

it('renders checked', () => {
  // First render
  const component = renderer.create(
    <SvgCheckbox checked />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});