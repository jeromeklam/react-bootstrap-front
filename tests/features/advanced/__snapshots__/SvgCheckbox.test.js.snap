// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checked 1`] = `
<div
  className="basic-svg-checkbox"
>
  <label
    className="label-cbx"
    for="cbx"
  >
    <input
      checked="checked"
      class="invisible"
      id="cbx"
      type="checkbox"
    />
    <div
      className="checkbox"
      style={
        Object {
          "height": "16px",
          "width": "16px",
        }
      }
    >
      <svg
        height="16px"
        viewBox="0 0 20 20"
        width="16px"
      >
        <path
          d="M3,1 L17,1 L17,1 C18.1045695,1 19,1.8954305 19,3 L19,17 L19,17 C19,18.1045695 18.1045695,19 17,19 L3,19 L3,19 C1.8954305,19 1,18.1045695 1,17 L1,3 L1,3 C1,1.8954305 1.8954305,1 3,1 Z"
        />
        <polyline
          points="4 11 8 15 16 6"
        />
      </svg>
    </div>
  </label>
</div>
`;

exports[`renders with className 1`] = `
<div
  className="basic-svg-checkbox"
>
  <label
    className="label-cbx"
    for="cbx"
  >
    <input
      class="invisible"
      id="cbx"
      type="checkbox"
    />
    <div
      className="checkbox"
      style={
        Object {
          "height": "16px",
          "width": "16px",
        }
      }
    >
      <svg
        height="16px"
        viewBox="0 0 20 20"
        width="16px"
      >
        <path
          d="M3,1 L17,1 L17,1 C18.1045695,1 19,1.8954305 19,3 L19,17 L19,17 C19,18.1045695 18.1045695,19 17,19 L3,19 L3,19 C1.8954305,19 1,18.1045695 1,17 L1,3 L1,3 C1,1.8954305 1.8954305,1 3,1 Z"
        />
        <polyline
          points="4 11 8 15 16 6"
        />
      </svg>
    </div>
  </label>
</div>
`;
